### PR TITLE
[build] Build Stevedore artifacts and have Yamato upload them (take 2)

### PR DESCRIPTION
**Pull request by kwi-dk**
This is a step towards having Bee download Tundra instead of bundling it, cutting 1.5 MB off the .exe size. This second iteration uses a .NET reimplementation of the Stevedore Upload tool, instead of the original Python version.

New build targets added:

artifacts/for-stevedore/tundra-linux-x64.zip
artifacts/for-stevedore/tundra-mac-x64.zip
artifacts/for-stevedore/tundra-win-x64.zip
artifacts/for-stevedore/tundra-win-x64-debug.7z

The individual Yamato per-platform "Build & Test" steps will now build and upload these artifacts.

The Yamato "Package Distribution" step will additionally summarize all uploaded artifacts:

    Stevedore artifacts uploaded:
    public: tundra-linux-x64/15d5db6de18f_70993d629517fb255218d1ba0863d305ac9e4a649ac584750ec3e888a5c33241.zip
    public: tundra-mac-x64/15d5db6de18f_de96949d333e866789bba8e166227116a3c6c11968dfd5e8573ace243b33996a.zip
    public: tundra-win-x64-debug/15d5db6de18f_6483d3d58c4558ec8a4e998f7bac7cdcbb2bcd27811db0eaa0dce8d82532963a.7z
    public: tundra-win-x64/15d5db6de18f_86806827df06aa967fc59b7f200cab9420f9f86a77a5d1f5be13ad6f14975cba.zip

Example results can be seen here:

https://yamato.prd.cds.internal.unity3d.com/jobs/250-tundra/tree/sorenl/.yamato%252Fyamato-ci.yml%2523package/594563/job

https://stevedore.unity3d.com/list/

### CODE REVIEW

File: .yamato/yamato-ci.yml [hunk 1]
```diff
@@ -9,6 +9,9 @@ linux:
     - git submodule update
     - mono bee.exe tundra2::linux::master
     - mono bee.exe tundra2-unittest-report::linux::master
+    - mono bee.exe artifacts/for-stevedore/tundra-linux-x64.zip
+    - curl -sSo StevedoreUpload.exe "$STEVEDORE_UPLOAD_TOOL_URL"
+    - mono StevedoreUpload.exe --append-manifest=build/linux-x86_64/manifest.stevedore --version-len=12 --repo=public --version="$GIT_REVISION" artifacts/for-stevedore/*
```
Comment (richard-fine) [inline .yamato/yamato-ci.yml:14]: Is that manifest.stevedore file something that you're asking StevedoreUpload to generate, or is it something that has been built by some build step outside this PR?

**Comment by richard-fine** [comment]
On https://stevedore.unity3d.com/list/ I see that the Tundra artefacts are marked as uploaded by 'sorenl+tundra@unity3d.com' - where is that coming from?

File: .yamato/yamato-ci.yml (hunk 1 above)
Comment (kwi-dk) [reply .yamato/yamato-ci.yml:14]: `--append-manifest` writes (appends) a line to the specified text file. In this case, the file is not pre-existing, so StevedoreUpload creates it. ([StevedoreUpload.exe --help](https://docs.google.com/document/d/1rfIP-Ncoj8OZUXhfBRyxLEIqIohXHL5ODjBP7KYY6PA/edit#heading=h.pmmsvxotsqs4))

The final step then concatenates the three individual manifests and displays them in the log. (The generated manifest file is not used for anything else.)

**Comment by kwi-dk** [comment]
'sorenl+tundra@unity3d.com' is the contact email associated with the [upload key](https://docs.google.com/document/d/1rfIP-Ncoj8OZUXhfBRyxLEIqIohXHL5ODjBP7KYY6PA/edit#heading=h.xn140l4hacw6).

**Comment by richard-fine** [comment]
Ah, so it's using an env var / secret group. OK, sounds good!

**Pull request merged**